interface/billing/ub04_form.php

```diff
@@ -0,0 +1,1404 @@
+              if ( term in cache ) {
+                response( cache[ term ] );
+                return;
+              }
+              $.getJSON( "./ub04_helpers.php", request, function( data, status, xhr ) {
+                cache[ term ] = data;
+                response( data );
+              });
+            }
+          }).dblclick(function(event) {
+              $(this).autocomplete('search'," ");
+          });
+    });
+    $( function() {
+        var cache = {};
+        $( "[title*='VALUE CODE']" ).autocomplete({
+          minLength: 1,
+          source: function( request, response ) {
+            var term = request.term;
+            request.code_group = "value_codes";
+            if ( term in cache ) {
+              response( cache[ term ] );
+              return;
+            }
+            $.getJSON( "./ub04_helpers.php", request, function( data, status, xhr ) {
+              cache[ term ] = data;
+              response( data );
+            });
+          }
+        }).dblclick(function(event) {
+            $(this).autocomplete('search'," ");
+        });
+      });
+    $( function() {
+        var cache = {};
+        $( "[title*='CONDITION CODE']" ).autocomplete({
+          minLength: 1,
+          source: function( request, response ) {
+            var term = request.term;
+            request.code_group = "ub_condition_code";
+            if ( term in cache ) {
+              response( cache[ term ] );
+              return;
+            }
+            $.getJSON( "./ub04_helpers.php", request, function( data, status, xhr ) {
+              cache[ term ] = data;
+              response( data );
+            });
+          }
+        }).dblclick(function(event) {
+            $(this).autocomplete('search'," ");
+        });
+      });
+    $( function() {
+        var cache = {};
+        $( "[title*='STATUS']" ).autocomplete({
+          minLength: 1,
+          source: function( request, response ) {
+            var term = request.term;
+            request.code_group = "patient_status_code";
+            if ( term in cache ) {
+              response( cache[ term ] );
+              return;
+            }
+            $.getJSON( "./ub04_helpers.php", request, function( data, status, xhr ) {
+              cache[ term ] = data;
+              response( data );
+            });
+          }
+        }).dblclick(function(event) {
+            $(this).autocomplete('search'," ");
+        });
+      });
+
+    $('form :input').change(function(e){
+           formChanged = true;
+           $(this).css("color","red");
+           return false;
+    });
+});
+<?php } else {?>
+<script type="text/javascript">
+<?php } ?>
+var pid;
+var encounter;
+var align = true;
+var isTemplate;
+var ub04id = new Array();
+payerid = '<?php echo $payerid;?>';
+pid = <?php echo $pid;?>;
+encounter = <?php echo $encounter;?>;
+isTemplate = <?php echo ($isAuthorized === true ? $isAuthorized : 0); ?>;
+ub04id = <?php echo $ub04id;?>;
```

Comment:
Escaping ub04id(a json encoded array of objects) will blow up the array. So a no there. Will escape the others but I never see the sense escaping variables that really don't require it. A boolean for example or integers.